cluster/cluster.go

```diff
@@ -23,12 +23,15 @@ type Cluster interface {
 	Member(filter map[string]string, status, name string) (Member, error)
 	MemberChan() chan []Member
 	ResolveNodeURL(streamURL string) (string, error)
+	EventChan() <-chan serf.UserEvent
+	BroadcastEvent(serf.UserEvent) error
 }
 
 type ClusterImpl struct {
 	config   *config.Cli
 	serf     *serf.Serf
-	eventCh  chan serf.Event
+	serfCh   chan serf.Event
+	eventCh  chan serf.UserEvent
```

Comment:
While you're in here, could you add some comments around what the channels are used for and the flow etc. please?

Reply:
Updated in 351b6385c825aceb2cf33c6897e8e88a3ee84076